src/components/Filters/index.tsx

```diff
@@ -111,25 +111,43 @@ class Filters extends React.Component<FiltersProps, FiltersState> {
 		const { filters, schema, views } = this.props;
 
 		const flatSchema = SchemaSieve.flattenSchema(schema);
+		const flatViews = this.flattenViews(views || []);
+		const flatFilters = (filters || []).map(filter =>
+			SchemaSieve.flattenSchema(filter),
+		);
 
 		this.state = {
 			showModal: false,
 			searchString: '',
 			editingFilter: null,
-			filters: filters || [],
-			views: views || [],
+			filters: flatFilters,
+			views: flatViews,
 			schema: flatSchema,
 			edit: [],
 		};
 
 		this.state.edit.push(this.getCleanEditModel());
 	}
 
-	componentWillReceiveProps(prevProps: FiltersProps) {
+	componentWillReceiveProps(nextProps: FiltersProps) {
 		// If the schema prop updates, also update the internal 'flat' schema
-		if (!isEqual(prevProps.schema, this.props.schema)) {
+		if (!isEqual(nextProps.schema, this.props.schema)) {
+			this.setState({
+				schema: SchemaSieve.flattenSchema(nextProps.schema),
+			});
+		}
+
+		if (nextProps.filters && !isEqual(nextProps.filters, this.props.filters)) {
 			this.setState({
-				schema: SchemaSieve.flattenSchema(this.props.schema),
+				filters: nextProps.filters.map(filter =>
+					SchemaSieve.flattenSchema(filter),
+				),
+			});
+		}
+
+		if (nextProps.views && !isEqual(nextProps.views, this.props.views)) {
```

Comment:
Because each of these checks check whether `views`/`filters` are truthy & different, not just different, with this change the state won't update at all if you remove the filters or views prop entirely.

Is that right? Sounds like a bug.

Reply:
You're right, that sounds like a bug. If there were filters/views and they get set to a falsey value, then we should set the state to an empty array, like we do in the constructor